HttpWebApi/HttpWebApi/Repositories/FileRepository.cs

```diff
@@ -0,0 +1,81 @@
+﻿namespace HttpWebApi.Repositories
+{
+    public class FileRepository
+    {
+
+        private Mutex fileMutex = new Mutex(false, "appMutex");
+        public Dictionary<Guid, Product> productsDictionary = new Dictionary<Guid, Product>
+        {
+
+        };
+
+        public async Task WriteToFileAsync(string fileName)
+        {
+            try
+            {
+                if (!fileMutex.WaitOne(TimeSpan.FromSeconds(5)))
```

Comment:
No longer necessary to use mutex, if you do not plan to open multiple instances of your app. File stream has a lock mechanism 'FileAccess.Write' already for one process